python/eos/analysis.py

```diff
@@ -155,6 +155,18 @@ def __init__(self, priors, likelihood, global_options={}, manual_constraints={},
         for n in varied_parameter_names - used_parameter_names:
             eos.warn('likelihood does not depend on parameter \'{}\'; remove from prior or check options!'.format(n))
 
+    def __sanitize_manual_input(self, data):
+        """Helper function that cast to float all entries of a manual_constraint"""
+        if type(data) is dict:
+            return { k: self.__sanitize_manual_input(v) for k, v in data.items() }
+        if not np.isscalar(data):
+            return list(map(self.__sanitize_manual_input, data))
```

Comment:
Hm. This is not correct. A ``dict`` also yields ``np.isscalar(dict()) == False``. I think we should explicitly handle only the classes we know. Or we use the ``collections`` module.

``` Python
from collections.abc import Collection, Mapping
...
if isinstance(data, Mapping):  # first check for mapping, i.e., dict-like
    return { self.__sanitize_manual_input(k): self.__sanitize_manual_input(v) for k, v in data.items() }
elif isinstance(data, Collection): # non-mapping collections, i.e., list-like
    return list(map(self.__sanitize_manual_input, data))
...
```

@pluegh can you check that? I've had trouble using ``collections.abc.Sequence`` to detect ``numpy.ndarray``, hence the ugliness above.

Reply:
I will cover it in unit tests, coming as soon as possible.

Reply:
It is correct since `dict` has been tested just before. But I agree that this is not elegant...